### PR TITLE
Add support for multiplayer auto countdown timers

### PR DESCRIPTION
**Pull request by smoogipoo**
Duration provided in seconds as I don't believe there's a time-span compatible MYSQL data type. Capped at SMALLINT as we're likely never going to support auto-start countdowns longer than 5 minutes.

### CODE REVIEW

File: database/migrations/2022_03_18_125356_add_auto_start_duration_to_multiplayer_rooms.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class AddAutoStartDurationToMultiplayerRooms extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('multiplayer_rooms', function (Blueprint $table) {
```
Comment (peppy) [inline database/migrations/2022_03_18_125356_add_auto_start_duration_to_multiplayer_rooms.php:16]: Hesitant to keep adding columns for each setting like this (we've moved to `json` storage for other similar cases). Fine this time to keep the changes simple, but just noting my concerns.

Comment (smoogipoo) [reply database/migrations/2022_03_18_125356_add_auto_start_duration_to_multiplayer_rooms.php:16]: Absolutely. And this column is being used for both playlists and multiplayer which have different requirements that are being bunched up together here.

That sort of change is a little too involved for me to make in quick order so I'll let someone else handle the web-side of things whenever that happens.

**Review by peppy — APPROVED** [review]

**Review by peppy — APPROVED** [review]

File: app/Models/Multiplayer/Room.php [hunk 2]
```diff
@@ -462,6 +462,7 @@ public function startGame(User $host, array $rawParams)
             'playlist:array',
             'type',
             'queue_mode',
+            'auto_start_duration:int',
```
Comment (notbakaneko) [inline app/Models/Multiplayer/Room.php:465]: Is this field going to be optional?
Does `0` mean "not set" or "start immediately"?

Comment (peppy) [reply app/Models/Multiplayer/Room.php:465]: 0 means no auto

**Pull request merged**